enforcer/src/main/java/org/wso2/choreo/connect/enforcer/api/APIFactory.java

```diff
@@ -64,10 +66,18 @@ public void addApis(List<Api> apis) {
         ConcurrentHashMap<String, API> newApis = new ConcurrentHashMap<>();
 
         for (Api api : apis) {
-            RestAPI enforcerApi = new RestAPI();
-            enforcerApi.init(api);
-            String apiKey = getApiKey(enforcerApi);
-            newApis.put(apiKey, enforcerApi);
+            if (api.getApiType().equals(APIConstants.ApiType.REST)) {
+                RestAPI enforcerApi = new RestAPI();
+                enforcerApi.init(api);
+                String apiKey = getApiKey(enforcerApi);
+                newApis.put(apiKey, enforcerApi);
+            } else {
```

Comment:
Lets reverse this condition, check if the type is ws, if not consider as a REST API

Reply:
fixed